src/Microsoft.Health.Fhir.CosmosDb/Features/Search/FhirCosmosSearchService.cs

```diff
@@ -40,10 +40,23 @@ public class FhirCosmosSearchService : SearchService
             SearchOptions searchOptions,
             CancellationToken cancellationToken)
         {
-            return await ExecuteSearchAsync(
+            SearchResult searchResult = await ExecuteSearchAsync(
                 _queryBuilder.BuildSqlQuerySpec(searchOptions),
                 searchOptions,
                 cancellationToken);
+
+            if (searchOptions.IncludeTotal == TotalType.Accurate && !searchOptions.CountOnly)
+            {
+                searchOptions.AsCountOnly(true);
```

Comment:
I think for Cosmos DB, this won't scale because it means for every request, it has to recompute the total count (even though that's might not be changing). Counting for Cosmos DB is also surprisingly slow, so that's another problem too.

Reply:
Here is an example you can try:

https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/17af5f40-c564-4afe-ada0-fe7193bd474a/resourceGroups/jackliu-ip2/providers/Microsoft.DocumentDb/databaseAccounts/jackliu-ip2/queryExplorer

I have about 3 million resources in this Cosmos DB. If you run the query `SELECT VALUE COUNT(1) FROM c WHERE c.isSystem = false AND c.isHistory = false AND c.isDeleted = false`, which is pretty close to what we would generate, then you will see that it dispatches multiple requests (each against one partition) and each of that costs like 3600RU, which quickly uses up all available RUs of the database and that can cause DDoS very easily.

I think for this to scale

Reply:
As discussed, here are my action items moving forward:

- Reach out to the Cosmos DB team to see if there is a better approach (done)
- For now, throw a Not Supported exception in FhirCosmosSearchService.cs if the user attempts to specify `_total = accurate` (still to be done)

Reply:
Implemented in 9830b25.

Reply:
.. adding this back in after all in d43999e.

Reasoning: the Cosmos team is rolling out changes that should greatly reduce the computation cost of the queries that we are executing. Thanks @jackliums for looking into this.